### ISSUE TITLE
Runtime error when trying to launch a Taurus widget without a Tango Host

### ISSUE THREAD

**Issue by LJBD** [taurus-org/taurus#528]
Hello!

I've just started reviewing #510 and I've stumbled across a weird issue.
When trying to launch a Taurus widget (I've tested it with `taurustrend`, `taurusform and `taurusplot`) without a `TANGO_HOST` env variable, I get the following error:

```python
(taurus_test) [ljbd@taurus-test ~]$ taurustrend
MainThread     INFO     2017-07-20 13:53:15,894 TaurusRootLogger: Using "PyQt4" for Qt
MainThread     INFO     2017-07-20 13:53:16,036 taurus.qt.qtgui.icon.icon: Setting Tango icon theme (from /home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/qt/qtgui/icon/)
Traceback (most recent call last):
  File "/home/ljbd/miniconda3/envs/taurus_test/bin/taurustrend", line 11, in <module>
    sys.exit(main())
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/qt/qtgui/plot/taurustrend.py", line 1951, in main
    w.showDataImportDlg()
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/qt/qtgui/plot/taurusplot.py", line 2794, in showDataImportDlg
    selectables=[taurus.core.taurusbasetypes.TaurusElementType.Attribute])
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/qt/qtgui/panel/taurusmodelchooser.py", line 167, in __init__
    host = taurus.Authority().getNormalName()
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/core/taurushelper.py", line 360, in Authority
    return Factory(getSchemeFromName(name or '')).getAuthority(name)
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/core/tango/tangofactory.py", line 246, in getAuthority
    self.dft_db = _Authority()
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/core/tango/tangodatabase.py", line 670, in __init__
    self.warning("Error getting default Tango host:", str(e))
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/core/util/log.py", line 835, in warning
    self.log_obj.warning(msg, *args, **kw)
  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/core/tango/tangodatabase.py", line 713, in __getattr__
    if not self.dbObj is None:

(... many, many errors in __getattr__ like that more...)

  File "/home/ljbd/miniconda3/envs/taurus_test/lib/python2.7/site-packages/taurus/core/tango/tangodatabase.py", line 713, in __getattr__
    if not self.dbObj is None:
RuntimeError: maximum recursion depth exceeded
```

That's really ugly...

Here's a file with the complete output: [taurus_recursion_error.txt](https://github.com/taurus-org/taurus/files/1162202/taurus_recursion_error.txt)

**Comment by mrosanes** [comment]
Bug #529 is also related with this bug.

**Comment by cpascual** [comment]
Just a pedantic note:

It works well without a TANGO_HOST env variable if you have the TANGO_HOST declared in /etc/tangorc (this is the default in debian)

**Comment by cpascual** [comment]
Closed via  bc114cf
